microsoft/testsuites/performance/nestedperf.py

```diff
@@ -585,61 +596,83 @@ def _storage_perf_qemu(
-        num_jobs = []
-        iodepth_iter = start_iodepth
-        core_count = node.tools[Lscpu].get_core_count()
-        while iodepth_iter <= max_iodepth:
-            num_jobs.append(min(iodepth_iter, core_count))
-            iodepth_iter = iodepth_iter * 2
+        l1_data_disk_count = len(l1_data_disks)
 
-        # run fio test
-        perf_disk(
-            l2_vm,
-            start_iodepth,
-            max_iodepth,
-            filename,
-            test_name=inspect.stack()[1][3],
-            core_count=core_count,
-            disk_count=l1_data_disk_count,
-            disk_setup_type=DiskSetupType.raid0,
-            disk_type=DiskType.premiumssd,
-            environment=environment,
-            num_jobs=num_jobs,
-            size_gb=8,
-            overwrite=True,
-        )
+        try:
+            # setup raid on l1 data disks
+            if setup_raid:
+                disks = ["/dev/md0"]
+                l1_partition_disks = reset_partitions(node, l1_data_disks)
+                stop_raid(node)
+                reset_raid(node, l1_partition_disks)
+            else:
+                disks = [l1_data_disks[0]]
+
+            # get l2 vm
+            l2_vm = qemu_connect_nested_vm(
+                node,
+                nested_image_username,
+                nested_image_password,
+                nested_image_port,
+                nested_image_url,
+                disks=disks,
+            )
+            l2_vm.capability.network_interface = Synthetic()
+
+            # Qemu command exits immediately but the VM requires some time to boot up.
+            l2_vm.tools[Lscpu].get_core_count()
+
+            # Each fio process start jobs equal to the iodepth to read/write from
+            # the disks. The max number of jobs can be equal to the core count of
+            # the node.
+            # Examples:
+            # iodepth = 4, core count = 8 => max_jobs = 4
+            # iodepth = 16, core count = 8 => max_jobs = 8
+            num_jobs = []
+            iodepth_iter = start_iodepth
+            core_count = node.tools[Lscpu].get_core_count()
+            while iodepth_iter <= max_iodepth:
+                num_jobs.append(min(iodepth_iter, core_count))
+                iodepth_iter = iodepth_iter * 2
+
+            # Run fio test
+            # The added disks appear as /dev/sdb on the nested vm
+            perf_disk(
+                l2_vm,
+                start_iodepth,
+                max_iodepth,
+                filename,
+                test_name=inspect.stack()[1][3],
+                core_count=core_count,
+                disk_count=l1_data_disk_count,
+                disk_setup_type=DiskSetupType.raid0,
+                disk_type=DiskType.premiumssd,
+                environment=environment,
+                num_jobs=num_jobs,
+                size_gb=8,
+                overwrite=True,
+            )
+        finally:
+            node.tools[Qemu].delete_vm()
```

Comment:
If it's shared by all cases in this class, it can be written in `after_case`.

Reply:
It's not. We have both hyperv and qemu testcases here. There are functions to stop VM in Qemu and Hyperv./ But maybe I can do a platform check here and call the functions accordingly